src/App.tsx

```diff
@@ -1,34 +1,72 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
-  return (
-    <>
-      <div className="section">
-        <div className="container">
-          <div className="box">
-            <h1 className="title">Todos:</h1>
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoadedTodos, setIsLoadedTodos] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [inputState, setInputState] = useState('');
+  const [toggleFilter, setToggleFilter] = useState<boolean | null>(null);
+
+  const fetchData = useCallback(async () => {
+    const data = await getTodos();
+
+    setTodos(data);
+    setIsLoadedTodos(true);
+  }, []);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
 
-            <div className="block">
-              <TodoFilter />
-            </div>
+  const filteredTodos = todos.filter(todo => {
+    const normInputState = inputState.toLocaleLowerCase().trim();
+    const isTitleMatch = todo.title.toLocaleLowerCase().includes(normInputState);
+    const isCompletedMatch = toggleFilter === null || todo.completed === toggleFilter;
```

Comment:
what do you mean `isTitleMatch/isCompletedMatch`? not really clear for me by its name